src/coreclr/unitygc/unitygc.cpp

```diff
@@ -0,0 +1,921 @@
+    }
+
+    // Sets whether or not a GC is in progress.
+    virtual void SetGCInProgress(bool fInProgress)
+    {
+        m_bGCInProgress = fInProgress;
+    }
+
+    // Gets whether or not the GC runtime structures are in a valid state for heap traversal.
+    virtual bool RuntimeStructuresValid()
+    {
+        return true;
+    }
+
+    // Tells the GC when the VM is suspending threads.
+    virtual void SetSuspensionPending(bool fSuspensionPending)
+    {
+    }
+
+    // Tells the GC how many YieldProcessor calls are equal to one scaled yield processor call.
+    virtual void SetYieldProcessorScalingFactor(float yieldProcessorScalingFactor)
+    {
+    }
+
+    // Flush the log and close the file if GCLog is turned on.
+    virtual void Shutdown()
+    {
+    }
+
+    /*
+    ============================================================================
+    Add/RemoveMemoryPressure support routines. These are on the interface
+    for now, but we should move Add/RemoveMemoryPressure from the VM to the GC.
+    When that occurs, these three routines can be removed from the interface.
+    ============================================================================
+    */
+
+    // Get the timestamp corresponding to the last GC that occured for the
+    // given generation.
+    virtual size_t GetLastGCStartTime(int generation)
+    {
+        assert(0);
+        return 0;
+    }
+
+    // Gets the duration of the last GC that occured for the given generation.
+    virtual size_t GetLastGCDuration(int generation)
+    {
+        assert(0);
+        return 0;
+    }
+
+    // Gets a timestamp for the current moment in time.
+    virtual size_t GetNow()
+    {
+        assert(0);
+        return 0;
+    }
+
+    /*
+    ===========================================================================
+    Allocation routines. These all call into the GC's allocator and may trigger a garbage
+    collection. All allocation routines return NULL when the allocation request
+    couldn't be serviced due to being out of memory.
+    ===========================================================================
+    */
+
+    // Allocates an object on the given allocation context with the given size and flags.
+    // It is the responsibility of the caller to ensure that the passed-in alloc context is
+    // owned by the thread that is calling this function. If using per-thread alloc contexts,
+    // no lock is needed; callers not using per-thread alloc contexts will need to acquire
+    // a lock to ensure that the calling thread has unique ownership over this alloc context;
+    virtual Object* Alloc(gc_alloc_context* acontext, size_t size, uint32_t flags)
+    {
+        const size_t k16K = 1ULL << 14;
+        size_t actualSize = size + sizeof(ObjHeader);
+        // align to 8 minimallly
+        actualSize = UGC_ALIGN(actualSize,sizeof(void*));
+
+        if (!acontext->alloc_ptr || acontext->alloc_ptr + actualSize > acontext->alloc_limit)
```

Comment:
Is it possible for `alloc_ptr` to be null? Otherwise wouldn't we nullref down on line 492?

Reply:
I think if `alloc_ptr` is `NULL` here, then it will be set on line 485. So line 492 should be safe.

Reply:
Nit: I would prefer the check be `if (acontext->alloc_ptr == NULL`...` to be the null check explicit though.